lib/ShareProvider.js

```diff
@@ -146,6 +150,44 @@ const Overlay = {
     if (window.SocialShare.shareButton) {
       window.SocialShare.shareButton.setAttribute("hidden", "true");
     }
+
+    // monkeypatch SocialActivationListener.receiveMessage
+    window.SocialActivationListener.originalReceiveMessage = window.SocialActivationListener.receiveMessage;
+    window.SocialActivationListener.receiveMessage = function(aMessage) {
+      // NOTE: This is a patched version of SocialActivationListener.receiveMessage
+      // that doesn't install the shareplane button to the toolbar when a new provider
+      // is activated.
+      // original code is at mozilla-central/source/browser/base/content/browser-social.js
+      let data = aMessage.json;
+      let browser = aMessage.target;
+      data.window = window;
+      const {
+        gBrowser,
+        Social,
+        SocialShare,
+        SocialSidebar
+      } = window;
+      // if the source if the message is the share panel, we do a one-click
+      // installation. The source of activations is controlled by the
+      // social.directories preference
+      let options;
+      if (browser === SocialShare.iframe && Services.prefs.getBoolPref("social.share.activationPanelEnabled")) {
+        options = {bypassContentCheck: true, bypassInstallPanel: true};
+      }
+
```

Comment:
The above code [170-176] is not necessary here since you're disabling the share panel.